src/coord/src/coord.rs

```diff
@@ -332,6 +334,49 @@ fn concretize_replica_config(
     Ok(config)
 }
 
+/// Holds tables needing advancement.
+struct AdvanceTables<T> {
+    map: HashMap<GlobalId, T>,
+}
+
+impl<T: CoordTimestamp> AdvanceTables<T> {
+    fn new() -> Self {
+        Self {
+            map: HashMap::new(),
+        }
+    }
+
+    // Insert an id that should be advanced to ts. If an existing and earlier entry
+    // exists, update it to ts.
+    fn insert(&mut self, id: GlobalId, ts: T) {
+        match self.map.entry(id) {
+            Entry::Occupied(mut entry) => {
+                if entry.get().less_than(&ts) {
+                    entry.insert(ts);
+                }
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(ts);
+            }
+        }
+    }
+
+    // Returns the next table to advance. Blocks forever if there are none.
+    async fn recv(&mut self) -> (GlobalId, T) {
```

Comment:
It might be nice to be able to toggle this to return multiple `GlobalId`s at once. Only advancing a single table at once seems like the opposite extreme of advancing all tables at once. The right tradeoff might be somewhere in the middle.

Reply:
This is subjective though, so feel free to disagree.

Reply:
persist changed their stuff to pipeline. It's not parallel, but it at least stops wait for responses before sending the next request, so things can be done much more in parellel. This PR would essentially negate all of that and overall increase the total amount of time it takes to close all tables. So I do think that there is *some* number of things we should be batching here.

The problem we are attempting to address is: an incoming query is blocked by table closing. We don't know when these queries will arrive. Let's say we have some target like: incoming queries are not delayed due to table closing by more than 50ms. This function could measure the time of its previous close attempt (or some weighted avg), and either increase or decrease the number of tables it attempts to close based on how long the previous attempt took relative to the target. This has some benefits like: if the pg server gets more loaded we back off, if it gets faster we speed up, if persist is able to parallelize those requests (say, via a connection pool), then we can aggressively speed up batching. This seems worth doing, I'll look at this.